sihl/src/database_migration.ml

```diff
@@ -220,10 +221,68 @@ struct
     execute steps
   ;;
 
+  let migrations_status () =
+    let open Lwt.Syntax in
+    let* migrations_states = Repo.get_all (table ()) in
+    let migration_states_namespaces =
+      migrations_states
+      |> List.map (fun migration_state ->
+             migration_state.Database_migration_repo.Migration.namespace)
+    in
+    let registered_migrations_namespaces =
+      Map.to_seq !registered_migrations |> List.of_seq |> List.map fst
+    in
+    let namespaces_to_check =
+      List.concat
+        [ migration_states_namespaces; registered_migrations_namespaces ]
+    in
+    Lwt.return
+    @@ List.map
+         (fun namespace ->
+           let migrations = Map.find_opt namespace !registered_migrations in
+           let migration_state =
+             List.find_opt
+               (fun migration_state ->
+                 String.equal
+                   migration_state.Database_migration_repo.Migration.namespace
+                   namespace)
+               migrations_states
+           in
+           match migrations, migration_state with
+           | None, None -> namespace, None
+           | None, Some migration_state ->
+             ( namespace
+             , Some (-migration_state.Database_migration_repo.Migration.version)
+             )
+           | Some migrations, Some migration_state ->
+             let unapplied_migrations_count =
+               List.length migrations
+               - migration_state.Database_migration_repo.Migration.version
+             in
+             namespace, Some unapplied_migrations_count
+           | Some migrations, None -> namespace, Some (List.length migrations))
+         namespaces_to_check
+  ;;
+
+  let pending_migrations () =
+    let open Lwt.Syntax in
+    let* unapplied = migrations_status () in
+    let rec find_pending result = function
+      | (namespace, Some n) :: xs ->
+        if n > 0
```

Comment:
If this check is done here, we don't need the `else if count < 0` in `check_migration_status`, correct?

Reply:
See comment above.

Reply:
Which comment? Not sure where you addressed this

Reply:
The one you are referring to, the one about rewriting it with `filter_map`.

Reply:
I don't see how that answers my question. The if-else-branch `else if count < 0` in the function `check_migration_status` becomes obsolete with this `n > 0` check, so we should delete that if-else-branch too.

Reply:
Ah sorry, now I see. `pending_migrations` is completely unrelated to that part. Nevermind!